src/components/MapSwitchHeader.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useCallback } from 'react';
+import { StyleSheet, TouchableOpacity, View } from 'react-native';
+import { colors, normalize } from '../config';
+import { BoldText } from './Text';
+
+type Props = {
+  setShowMap: (newValue: boolean) => void,
+  showMap: boolean
+}
+
+export const MapSwitchHeader = ({ setShowMap, showMap }: Props) => {
+  const onPressShowMap = useCallback(() => setShowMap(true), [setShowMap]);
+  const onPressShowList = useCallback(() => setShowMap(false), [setShowMap]);
+
+  return <View style={styles.container}>
+    <TouchableOpacity onPress={onPressShowList} style={styles.button}>
```

Comment:
we have a `Touchable` component that should be used here, to use the correct components depending on the operating system. i will add that as well when adding the new component for styling.

Reply:
this introduced a typescript issue which I resolved with 8e4977015a7e291a01c41326667346eedfb06dbb

Reply:
thanks. i tried but wasn't able to find a solution.
we need to check if your rewriting works everywhere. have you checked all screens, where `Touchable` is used?
is something like this still working, where a component is passed as prop: `Component={Touchable}`?

Reply:
I double checked everything on iOS device and android emulator (my test device is still not available).

it is now a very simple functional component instead of a union type of two functional components that use a different set of props.

Reply:
all right, should work, thanks